test/emulation.spec.js

```diff
@@ -114,4 +114,20 @@ module.exports.addTests = function({testRunner, expect, puppeteer}) {
       expect(error.message).toBe('Unsupported media type: bad');
     });
   });
+
+  describe('Page.emulateTimezone', function() {
+    it('should work', async({page, server}) => {
+      await page.emulateTimezone('America/Jamaica');
+      expect(await page.evaluate(() => new Date(1479579154987).toString())).toBe('Sat Nov 19 2016 13:12:34 GMT-0500 (Eastern Standard Time)');
+
+      await page.emulateTimezone('Pacific/Honolulu');
+      expect(await page.evaluate(() => new Date(1479579154987).toString())).toBe('Sat Nov 19 2016 08:12:34 GMT-1000 (Hawaii-Aleutian Standard Time)');
+
+      await page.emulateTimezone('America/Argentina/Buenos_Aires');
+      expect(await page.evaluate(() => new Date(1479579154987).toString())).toBe('Sat Nov 19 2016 15:12:34 GMT-0300 (Argentina Standard Time)');
+
+      await page.emulateTimezone('Europe/Berlin');
+      expect(await page.evaluate(() => new Date(1479579154987).toString())).toBe('Sat Nov 19 2016 19:12:34 GMT+0100 (Central European Standard Time)');
+    });
+  });
```

Comment:
maybe a test for passing an invalid tzId and verifying the rejection msg?

Reply:
I agree rejecting on invalid inputs would be useful. However, the CDP API doesn't signal whether the time zone is valid or not --- it just silently ignores invalid inputs.

![](https://user-images.githubusercontent.com/81942/65133926-c9fbf300-da03-11e9-98dd-405b5e245de5.png)

Any ideas? We could hardcode a list of recognized time zones in Puppeteer and reject on unknown inputs (similar to what we do for e.g. `page.emulateMedia`), but that would mean we'd have to maintain the list.

Reply:
We could also change the CDP to return some kind of signal if the input wasn't recognized, but that seems to deviate from other `set*Override` APIs.

Reply:
CDP has now been updated accordingly: https://chromium-review.googlesource.com/c/chromium/src/+/1822557 Once this change rolls into Puppeteer's `chromium_revision` we can make use of the negative signal to detect the failure case.

Reply:
Patch updated accordingly. It's blocked on a Chromium roll to v79.0.3923.0+, though.